src/dbos-executor.ts

```diff
@@ -1898,12 +1898,39 @@ export class DBOSExecutor implements DBOSExecutorContext {
     }
   }
 
-  getWorkflowStatus(workflowID: string, callerID?: string, callerFN?: number): Promise<WorkflowStatus | null> {
-    return this.systemDatabase.getWorkflowStatus(workflowID, callerID, callerFN);
+  async getWorkflowStatus(workflowID: string, callerID?: string, callerFN?: number): Promise<WorkflowStatus | null> {
+    const status = await this.systemDatabase.getWorkflowStatus(workflowID, callerID, callerFN);
+    const inputs = await this.systemDatabase.getWorkflowInputs(workflowID);
+    if (status) {
```

Comment:
Should we use the `toWorkflowStatus` below? Also, I noticed that `systemDatabase.getWorkflowStatus` doesn't get output or error, so they are always `null`. Should we change `systemDatabase.getWorkflowStatus` to also select output and error?

Reply:
This getWorkflowStatus specifically omitted the output and error fields because it is mainly used to check workflow pending status. In that case, we should be explicit that those fields are always undefined (maybe even add it to JSDoc above).

Reply:
Also, I don't think we need to get workflow inputs here if we're not getting workflow output or error.

Reply:
Looking at the Python code, we actually use listWorkflows(workflow_id=xxx) to return everything (including the input, output/error) of a specific workflow.

Reply:
There are a few places in sysdb where we make multiple round trips like this where we should optimize. Would like to do that in a separate PR

Reply:
Sure we can optimize it later, but `systemDatabase.getWorkflowStatus` is not retrieving the output/error. So the return status is incomplete.

Reply:
I'd like to handle this in a separate PR after #871 is merged.

Reply:
It would be easy enough for `sysdb.getWorkflowStatus` to call `sysdb.listWorkflows` under the hood, retrieving the status and inputs in one roundtrip. Every place we call `sysdb.getWorkflowStatus` today is followed by a call to `sysdb.getWorkflowInputs`, so this would save a DB round trip in several places. However, #871 changes the way systemdb handles workflow inputs, so it would make sense to do that work in a separate PR after this PR and #871 are both merged.

Reply:
As per slack convo, I updated `sysdb.getWorkflowStatus` to use `sysdb.listWorkflows` under the hood, so it returns the full set of status fields. Then I modified `executor.getWorkflowStatus` to use `toWorkflowStatus`. 

Note, there are a few other places where we call `sysdb.getWorkflowStatus` and then `sysdb.getWorkflowInputs` which is now redundant. However, as per earlier comment #871 changes the way `getWorkflowInputs` is handled so I added `TODO` comments to remove those redundant calls in a future PR.